Gem/Code/Source/Components/NetworkAiComponent.cpp

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project. For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <Source/Components/NetworkAiComponent.h>
+
+#include <AzCore/Serialization/EditContext.h>
+#include <AzCore/Serialization/SerializeContext.h>
+#include <Source/Components/WasdPlayerMovementComponent.h>
+#include <Source/Components/NetworkWeaponsComponent.h>
+#include <AzCore/Time/ITime.h>
+#include <Multiplayer/Components/NetBindComponent.h>
+
+namespace MultiplayerSample
+{
+    void NetworkAiComponent::Reflect(AZ::ReflectContext* context)
+    {
+        if (AZ::SerializeContext* serializeContext = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serializeContext->Class<NetworkAiComponent, NetworkAiComponentBase>()->Version(1);
+        }
+
+        NetworkAiComponentBase::Reflect(context);
+    }
+
+    void NetworkAiComponent::OnInit()
+    {
+    }
+
+    void NetworkAiComponent::OnActivate([[maybe_unused]] Multiplayer::EntityIsMigrating entityIsMigrating)
+    {
+    }
+
+    void NetworkAiComponent::OnDeactivate([[maybe_unused]] Multiplayer::EntityIsMigrating entityIsMigrating)
+    {
+    }
+
+    void NetworkAiComponent::TickMovement(WasdPlayerMovementComponentController& movementController, float deltaTime)
+    {
+        // TODO: Execute this tick only if this component is owned by this endpoint (currently ticks on server only)
+        m_remainingTimeMs -= deltaTime * 1000;
+
+        if (m_remainingTimeMs <= 0)
+        {
+            // Determine a new directive after 500 to 9500 ms
+            m_remainingTimeMs = m_lcg.GetRandomFloat() * (m_actionIntervalMaxMs - m_actionIntervalMinMs) + m_actionIntervalMinMs;
+            m_turnRate = 1.f / m_remainingTimeMs;
+            m_targetYawDelta = -movementController.m_viewYaw + (m_lcg.GetRandomFloat() * 2.f - 1.f);
+            m_targetPitchDelta = -movementController.m_viewPitch + (m_lcg.GetRandomFloat() - 0.5f);
+            m_action = static_cast<Action>(m_lcg.GetRandom() % static_cast<int>(Action::COUNT));
+            m_strafingRight = static_cast<bool>(m_lcg.GetRandom() % 2);
+        }
+
+        // Translate desired motion into inputs
+
+        // Interpolate the current view yaw and pitch values towards the desired values
+        movementController.m_viewYaw += 1000.f * m_turnRate * deltaTime * m_targetYawDelta;
```

Comment:
nit: should 1000.f be a constexpr somewhere here?

Reply:
This converts seconds to ms (since deltaTime in our engine is typically a float of seconds). Is there already a constant somewhere I should use, or would you prefer a locally defined constant?